systemtest/src/test/java/io/strimzi/systemtest/operators/PodSetIsolatedST.java

```diff
@@ -0,0 +1,115 @@
+
+        Map<String, String> kafkaPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getKafkaSelector());
+        Map<String, String> zkPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getZookeeperSelector());
+
+        LOGGER.info("Changing Kafka resource configuration, the pods should not be rolled");
+
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> kafka.getSpec().getZookeeper().setReadinessProbe(new ProbeBuilder().withTimeoutSeconds(probeTimeoutSeconds).build()), testStorage.getNamespaceName());
+        RollingUpdateUtils.waitForNoRollingUpdate(testStorage.getNamespaceName(), testStorage.getZookeeperSelector(), zkPods);
+
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> kafka.getSpec().getKafka().setReadinessProbe(new ProbeBuilder().withTimeoutSeconds(probeTimeoutSeconds).build()), testStorage.getNamespaceName());
+        RollingUpdateUtils.waitForNoRollingUpdate(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), kafkaPods);
+
+        LOGGER.info("Deleting one Kafka and one ZK pod, the should be recreated");
+        kubeClient().deletePodWithName(testStorage.getNamespaceName(), KafkaResources.kafkaPodName(testStorage.getClusterName(), 0));
+        PodUtils.waitForPodsReady(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), replicas, true);
+
+        kubeClient().deletePodWithName(testStorage.getNamespaceName(), KafkaResources.zookeeperPodName(testStorage.getClusterName(), 0));
+        PodUtils.waitForPodsReady(testStorage.getNamespaceName(), testStorage.getZookeeperSelector(), replicas, true);
+
+        kafkaPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getKafkaSelector());
+        zkPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getZookeeperSelector());
+
+        LOGGER.info("Removing {} env from CO", Environment.STRIMZI_POD_SET_RECONCILIATION_ONLY_ENV);
+
+        envVars.remove(reconciliationEnv);
+        DeploymentResource.replaceDeployment(Constants.STRIMZI_DEPLOYMENT_NAME,
+            coDep -> coDep.getSpec().getTemplate().getSpec().getContainers().get(0).setEnv(envVars), INFRA_NAMESPACE);
+
+        DeploymentUtils.waitTillDepHasRolled(INFRA_NAMESPACE, STRIMZI_DEPLOYMENT_NAME, 1, coPod);
+
+        LOGGER.info("Because the configuration was changed, pods should be rolled");
+
+        RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), testStorage.getZookeeperSelector(), replicas, kafkaPods);
+        RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), replicas, zkPods);
+    }
+
+    @BeforeAll
+    void setup() {
+        clusterOperator.unInstall();
+        clusterOperator = new SetupClusterOperator.SetupClusterOperatorBuilder()
```

Comment:
instead of pure instantiating with the `new` keyword, you can use something like this:
```java
clusterOperator = clusterOperator
            .defaultInstallation()
            .withExtraEnvVars(INITIAL_ENV_VARS)
            .createInstallation()
            .runInstallation();
```
the context and namespace will be fulfilled in `.defaultInstallation` step.